app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-pickup.js

```diff
@@ -0,0 +1,216 @@
+                function(rate) {
+                    return (
+                        rate.carrier_code === this.rate.carrier_code &&
+                        rate.method_code === this.rate.method_code
+                    );
+                },
+                this
+            );
+
+            this.preselectLocation();
+            this.selectShippingMethod(pickupShippingMethod);
+        },
+        /**
+         * @param {Object} shippingMethod
+         */
+        selectShippingMethod: function(shippingMethod) {
+            selectShippingMethodAction(shippingMethod);
+            checkoutData.setSelectedShippingAddress(
+                quote.shippingAddress().getKey()
+            );
+        },
+        convertAddressType: function(shippingAddress) {
+            if (
+                !this.isStorePickupAddress(shippingAddress) &&
+                this.isStorePickupSelected()
+            ) {
+                quote.shippingAddress(
+                    $.extend({}, shippingAddress, {
+                        canUseForBilling: function() {
+                            return false;
+                        },
+                        getType: function() {
+                            return 'store-pickup-address';
+                        },
+                    })
+                );
+            }
+        },
+        preselectLocation: function() {
+            if (!this.isStorePickupSelected()) {
+                return;
+            }
+
+            var selectedLocation = pickupLocationsService.selectedLocation();
+            if (selectedLocation) {
+                pickupLocationsService.selectForShipping(selectedLocation);
+                return;
+            }
+
+            var shippingAddress = quote.shippingAddress();
+            var customAttributes = shippingAddress.customAttributes || [];
+            var selectedSourceCode = _.findWhere(customAttributes, {
+                attribute_code: 'sourceCode',
+            });
+
+            if (selectedSourceCode) {
+                pickupLocationsService
+                    .getLocation(selectedSourceCode.value)
+                    .then(function(location) {
+                        pickupLocationsService.selectForShipping(location);
+                    });
+            } else if (shippingAddress.city && shippingAddress.postcode) {
+                pickupLocationsService
+                    .getNearbyLocations({
+                        radius: this.nearbySearchRadius,
+                        pageSize: this.nearbySearchLimit,
+                        country: shippingAddress.countryId,
+                        city: shippingAddress.city,
+                        postcode: shippingAddress.postcode,
+                        region: shippingAddress.region,
+                        country: this.defaultCountry,
+                    })
+                    .then(function(locations) {
+                        var nearestLocation = locations[0];
```

Comment:
What if the location array will be empty? Maybe, you should add some empty result text.